text/0000-wallet-replay-protection.md

```diff
@@ -0,0 +1,298 @@
+
+- Title: `wallet-(re)play-protection`
+- Authors: Phyro, John Tromp, Yeastplume
+- Start date: Aug 9, 2020
+- RFC PR: Edit if merged: [mimblewimble/grin-rfcs#0000](https://github.com/mimblewimble/grin-rfcs/pull/0000) 
+- Tracking issue: [Edit if merged with link to tracking github issue]
+
+---
+
+# Summary
+[summary]: #summary
+
+A few months ago, a new class of attacks was found on Grin that we call (re)play attacks. A Mimblewimble transaction that already happened can be replayed if the exact conditions are recreated. In order to be able to replay a transaction the following conditions must be true:
+1. The inputs must be in the utxo set
+2. The outputs of the transaction must not be in the utxo set (Grin does not allow duplicate outputs in the utxo set)
+
+This means that if Alice sent some coins to Bob and the outputs have been spent, anyone that saw their original transaction could replay the transaction if the same inputs existed in the utxo set. But why would the same inputs exist on the chain in the first place? While this seems harmless at first, it can with some creativity and careful coordination be used take someone else's coins without their permission. The attack is not easy to pull off, but is doable in some scenarios. Similarly, a play attack comes from the same reasoning, but with the difference that a transaction never made it to the chain for some reason. One such reason could be that an input was spent before it was broadcasted which would make the current transaction invalid. In this document, we propose new wallet behaviours that make it robust in the face of (re)play attacks so the end users can't be victims of these attacks.

```

Comment:
Currently, the summary focuses on the attacks that were discovered, and then briefly mentions what the purpose of the document is (to propose new wallet behaviours that make it robust in the face of these attacks).

Instead, I'd refocus the summary and make it much more high level and to the point: 

* Given certain conditions, UTXOs that belong to a user can be moved without their explicit permissions.
* This document outlines the fixes that prevents this behaviour from occurring, specifically x, y, and z.